File: Solutions/CPP/Majority Element II.cpp
#include <vector>
#include <map>
#include <iostream>
using namespace std ;

class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        vector<int> pairs;
        map<int, int> map;
        for (int num : nums) {
            map[num]++;
        }
        for (auto& [num, count] : map) {
            if (count > nums.size() / 3) {
                pairs.push_back(num);
            }
        }
        return pairs;
    }
};

int main () {
    Solution sol;
    vector<int> nums = {3,2,3};
    vector<int> result = sol.majorityElement(nums);
    for (int num : result) {
        cout << num << " ";
    }
    cout << endl;
    return 0;
}

File: Solutions/RUST/Majority Element II.rs
use std::collections::HashMap;

struct Solution;

impl Solution {
    fn majority_element(nums: Vec<i32>) -> Vec<i32> {
        let mut result = Vec::new();
        let mut freq: HashMap<i32, i32> = HashMap::new();

        // Count occurrences
        for num in &nums {
            *freq.entry(*num).or_insert(0) += 1;
        }

        // Collect elements appearing more than n/3 times
        for (num, count) in freq {
            if count > (nums.len() / 3) as i32 {
                result.push(num);
            }
        }

        result
    }
}

fn main() {
    let nums = vec![3, 2, 3];
    let result = Solution::majority_element(nums);

    for num in result {
        print!("{} ", num);
    }
    println!();
}
